### ISSUE TITLE
Provide pre-commit checks to ensure everything is likely to work

### ISSUE THREAD

**Issue by rgardler** [Azure/Moodle#7]
We can provide a bunch of pre-commit checks to catch common errors and, in some cases, provide scripts to help.

- [ ] Build a script to check the deploy, visualize, nested template and script URLs - reject commit if they don't match the git branch
  - This also needs to be run on pre-merge via the web UI, otherwise we run the risk of master not having the correct values
- [ ] Build a script to automatically update the deploy, visualize, nested template and script URLs - reject commit if they don't match the git branch

**Comment by rgardler** [comment]
@hosungsmsft you might appreciate the script in PR #7. It's not wired into pre-commit yet, but it is faster than manually checking.

**Comment by hosungsmsft** [comment]
@rgardler yes, and I was looking at your PR for review, but you already merged. I didn't understand in your script why the current branch is not embedded in the variable SCRIPT_LOCATION (contrast to BASE_TEMPLATE_URL), and you are comparing the BASE_TEMPLATE_URL with the scriptLocation value? Are they bugs?

**Comment by rgardler** [comment]
I need to document review processes (it’s actually in the checklist for this issue). Short version is that I believe in Commit then Review where appropriate. That is, if the code is not in the critical path and/or poses no risk then just merge it. People can review after the merge. This is a must faster process, because it’s often faster for a reviewer to make the change than it is to ask for the change.

Your observation below is a bug (I’m trying to move too fast!) If you have the time / inclination just fix it and merge your change. If you think I might be smarter than I look and have a good reason then raise it as a question as you have done here. But I can confirm they are bugs and I am not smarter than I look :-P

Ross

**Comment by hosungsmsft** [comment]
No worries, and you know I made an embarrassing bug myself (which I think led you to craft this), and fixing it urgently... Let me fix these bugs over the weekend, with broken links in other documentation. I'm right now fixing the Apache template blunder... :(
